internal/client/v1/client/vald/vald.go

```diff
@@ -156,16 +156,18 @@ func (c *client) StreamSearch(
 			span.End()
 		}
 	}()
-	_, err = c.c.RoundRobin(ctx, func(ctx context.Context,
+	_res, err := c.c.RoundRobin(ctx, func(ctx context.Context,
 		conn *grpc.ClientConn,
 		copts ...grpc.CallOption,
 	) (any, error) {
-		res, err = vald.NewValdClient(conn).StreamSearch(ctx, append(copts, opts...)...)
-		return nil, err
+		return vald.NewValdClient(conn).StreamSearch(ctx, append(copts, opts...)...)
 	})
 	if err != nil {
 		return nil, err
 	}
+	// err == nil, res should not be nil
+	// TODO: refactor RoundRobin using generics
+	res, _ = _res.(vald.Search_StreamSearchClient)
```

Comment:
It would be better to check _res is nil or not.
```suggestion
	res, _ = _res.(vald.Search_StreamSearchClient)
```

Reply:
Handled in `grpc.RoundRobin` function 🙏 
Thank you for your suggestion!